src/wp-content/themes/twentyfifteen/js/functions.js

```diff
@@ -5,130 +5,162 @@
  * Contains handlers for navigation and widget area.
  */
 
-( function( $ ) {
-	var $body, $window, $sidebar, resizeTimer,
-		secondary, button;
+( function() {
+	function domReady( callback ) {
+		if ( document.readyState === 'complete' || document.readyState === 'interactive' ) {
+			callback();
+		} else {
+			document.addEventListener( 'DOMContentLoaded', callback );
+		}
+	}
+
+	function toggleClass( element, className ) {
+		if ( -1 !== element.className.indexOf( className ) ) {
+			element.className = element.className.replace( ' ' + className, '' );
+		} else {
+			element.className += ' ' + className;
+		}
+	}
+
+	var sidebar, resizeTimer, secondary, button;
 
 	function initMainNavigation( container ) {
+		if ( ! container ) {
+			return;
+		}
+
 		// Add dropdown toggle that display child menu items.
-		container.find( '.menu-item-has-children > a' ).after( '<button class="dropdown-toggle" aria-expanded="false">' + screenReaderText.expand + '</button>' );
+		Array.from( container.querySelectorAll( '.menu-item-has-children > a' ) ).forEach( function( menuLink ) {
+			var dropdownToggle = document.createElement( 'button' );
+			dropdownToggle.className = 'dropdown-toggle';
+			dropdownToggle.setAttribute( 'aria-expanded', 'false' );
+			dropdownToggle.innerHTML = screenReaderText.expand;
+			menuLink.parentElement.insertBefore( dropdownToggle, menuLink.nextElementSibling );
+		} );
 
 		// Toggle buttons and submenu items with active children menu items.
-		container.find( '.current-menu-ancestor > button' ).addClass( 'toggle-on' );
-		container.find( '.current-menu-ancestor > .sub-menu' ).addClass( 'toggled-on' );
-
-		container.find( '.dropdown-toggle' ).on( 'click', function( e ) {
-			var _this = $( this );
-			e.preventDefault();
-			_this.toggleClass( 'toggle-on' );
-			_this.next( '.children, .sub-menu' ).toggleClass( 'toggled-on' );
-			_this.attr( 'aria-expanded', _this.attr( 'aria-expanded' ) === 'false' ? 'true' : 'false' );
-			_this.html( _this.html() === screenReaderText.expand ? screenReaderText.collapse : screenReaderText.expand );
+		Array.from( container.querySelectorAll( '.current-menu-ancestor > button, .current-menu-ancestor > .sub-menu' ) ).forEach( function( subItem ) {
+			subItem.className += 'toggle-on';
+		} );
+
+		Array.from( container.querySelectorAll( '.dropdown-toggle' ) ).forEach( function( dropdownToggle ) {
+			dropdownToggle.addEventListener( 'click', function( e ) {
+				e.preventDefault();
+				toggleClass( this, 'toggle-on' );
+				if ( this.nextElementSibling.tagName === 'UL' || this.nextElementSibling.tagName === 'OL' ) {
+					toggleClass( this.nextElementSibling, 'toggled-on' );
+				}
+				this.setAttribute( 'aria-expanded', this.getAttribute( 'aria-expanded' ) === 'false' ? 'true' : 'false' );
+				this.innerHTML = this.innerHTML === screenReaderText.expand ? screenReaderText.collapse : screenReaderText.expand;
```

Comment:
```suggestion
				dropdownToggle.textContent = dropdownToggle.textContent === screenReaderText.expand ? screenReaderText.collapse : screenReaderText.expand;
```
... assuming that `screenReaderText.expand` / `screenReaderText.collapse` only ever contain text and no HTML.

Reply:
See above, it has to support HTML.